src/widget.ts

```diff
@@ -384,53 +384,54 @@ export class DSVEditor extends Widget {
+    }
+
     const delimiter = this.delimiter;
-    const model = this._grid.dataModel as EditorModel;
-    let data: string;
-
-    if (!model) {
-      data = this._context.model.toString();
-      const dataModel = (this._grid.dataModel = new EditorModel({
-        data,
-        delimiter
-      }));
-      const selectionModel = new GhostSelectionModel({ dataModel });
-      this._grid.selectionModel = selectionModel;
-
-      // create litestore
-      this._litestore = new Litestore({
-        id: 0,
-        schemas: [DSVEditor.DATAMODEL_SCHEMA]
-      });
+    const data = this._context.model.toString();
+    const dataModel = (this._grid.dataModel = new EditorModel({
+      data,
+      delimiter
+    }));
+
+    this._grid.selectionModel = new GhostSelectionModel({ dataModel });
+
+    // create litestore
+    this._litestore = new Litestore({
+      id: 0,
+      schemas: [DSVEditor.DATAMODEL_SCHEMA]
+    });
 
-      // Give the litestore as a property of the model for it to read from.
-      this.dataModel.litestore = this._litestore;
-
-      // Define the initial update object for the litestore.
-      const update: DSVEditor.ModelChangedArgs = {};
-
-      // Define the initial state of the row and column map.
-      const rowUpdate = {
-        index: 0,
-        remove: 0,
-        values: toArray(range(0, this.dataModel.totalRows))
-      };
-      const columnUpdate = {
-        index: 0,
-        remove: 0,
-        values: toArray(range(0, this.dataModel.totalColumns))
-      };
-
-      // Add the map updates to the update object.
-      update.rowUpdate = rowUpdate;
-      update.columnUpdate = columnUpdate;
-
-      // set inital status of litestore
-      this.updateModel(update);
-      dataModel.onChangedSignal.connect(this._onModelSignal, this);
-      dataModel.isDataFormattedChanged.connect(this._updateRenderer, this);
-      // dataModel.cancelEditingSignal.connect(this._cancelEditing, this);
-    }
+    // Give the litestore as a property of the model for it to read from.
+    dataModel.litestore = this._litestore;
+
+    // Define the initial update object for the litestore.
+    const update: DSVEditor.ModelChangedArgs = {};
+
+    // Define the initial state of the row and column map.
+    const rowUpdate = {
+      index: 0,
+      remove: 0,
+      values: toArray(range(0, this.dataModel.totalRows))
+    };
+    const columnUpdate = {
+      index: 0,
+      remove: 0,
+      values: toArray(range(0, this.dataModel.totalColumns))
+    };
+
+    // Add the map updates to the update object.
+    update.rowUpdate = rowUpdate;
+    update.columnUpdate = columnUpdate;
+
+    // set inital status of litestore
+    this.updateModel(update);
+
+    // Connect to the the model signals.
+    dataModel.onChangedSignal.connect(this._onModelSignal, this);
+    dataModel.isDataFormattedChanged.connect(this._updateRenderer, this);
 
     // Update the div elements of the grid.
     this._updateContextElements();
```

Comment:
I think this needs to go above the bail statement, otherwise, this will never run after the first time

Reply:
That's a good point. I almost just want to put in in the constructor if possible because it really only needs to run that first time. `_updateGrid` only gets called on init and on a save but the elements are being kept up to date with resizing already.

Reply:
Okay that's actually tricky because if I put it at the beginning I think that will have unintended side effects because the grid sizes won't be correct so the div sizes will be off. And I think same goes for putting it in the constructor. Maybe just leave it for now?

Reply:
Yes, it checks out to me. We'll keep it there for now.